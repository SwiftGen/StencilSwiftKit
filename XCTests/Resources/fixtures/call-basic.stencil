{% set p1 %}1{% endset %}
{% set p2 %}hey{% endset %}
{% set p3 %}1234{% endset %}
{% set globalVarA %}Hello World!{% endset %}

// define test
{% macro test %}
  Hello world! (inside test)
  globalVarA = "{{globalVarA}}"
{% endmacro %}

// define test2
{% macro test2 %}
  Hello universe! (inside test2)
{% endmacro %}

// define test3
{% macro test3 a b c %}
  Received parameters in test3:
  - a = "{{a}}"
  - b = "{{b}}"
  - c = "{{c}}"

  // calling test2
  {% call test2 %}
{% endmacro %}

// calling test
{% call test %}

// calling test2
{% call test2 %}

// calling test3
{% call test3 p1 p2 p3 %}
