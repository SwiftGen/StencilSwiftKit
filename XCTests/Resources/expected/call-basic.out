

// define test

// define test2

// define test3

// calling test
  Hello world! (inside test)
  globalVarA = "Hello World!"

// calling test2
  Hello universe! (inside test2)

// calling test3
  Received parameters in test3:
  - a = "1"
  - b = "hey"
  - c = "1234"

  // calling test2
  Hello universe! (inside test2)
